plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -190,6 +204,84 @@ public TableSnapshot loadSnapshot(ConnectorSession session, SchemaTableName tabl
         return snapshot;
     }
 
+    private TableSnapshot loadSnapshotForTimeTravel(TrinoFileSystem fileSystem, SchemaTableName table, String tableLocation, long endVersion)
+            throws IOException
+    {
+        try {
+            Optional<LastCheckpoint> lastCheckpoint = findCheckpoint(fileSystem, tableLocation, endVersion);
+            return TableSnapshot.load(
+                    table,
+                    lastCheckpoint,
+                    fileSystem,
+                    tableLocation,
+                    parquetReaderOptions,
+                    checkpointRowStatisticsWritingEnabled,
+                    domainCompactionThreshold,
+                    Optional.of(endVersion));
+        }
+        catch (UncheckedExecutionException e) {
+            throwIfUnchecked(e.getCause());
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Optional<LastCheckpoint> findCheckpoint(TrinoFileSystem fileSystem, String tableLocation, Long endVersion)
+    {
+        Optional<LastCheckpoint> lastCheckpoint = readLastCheckpoint(fileSystem, tableLocation);
+        if (lastCheckpoint.isPresent() && lastCheckpoint.get().version() <= endVersion) {
+            return lastCheckpoint;
```

Comment:
is it possible to endVersion to be greater than lastCheckpoint? Let's throw in this case instead.

Reply:
Can we say as of version 25?

Reply:
No, time travel to such version should fail.